Eiffel curl updated:

jvelilla opened the pull request:
Added support to send mime emails.
Added example to show how to use it.

jvelilla committed:
Eiffel curl updated:
Added support to send mime emails.
Added example to show how to use it.

jocelyn commented on Src/examples/cURL/smtp_mime/smtp_mime.ecf:
```diff
@@ -0,0 +1,25 @@
+ xsi:schemaLocation="http://www.eiffel.com/developers/xml/configuration-1-16-0 http://www.eiffel.com/developers/xml/configuration-1-16-0.xsd" name="smtp_mime">
+	<target name="smtp_mime">
+		<root cluster="smtp_mime" class="APPLICATION" feature="make"/>
+		<option warning="true" is_obsolete_routine_type="true" syntax="transitional">
```
I guess you can remove the part ` is_obsolete_routine_type="true" syntax="transitional" `

jvelilla replied:
done

jocelyn commented on Src/library/cURL/curl_opt_constants.e:
```diff
@@ -801,6 +801,16 @@ feature -- Protocol
 		end
 
+	curlopt_mimepost: INTEGER
+			-- Declared as CURLOPT_TIMEVALUE
```
the comment mentions CURLOPT_TIMEVALUE ... but I guess this is a copy/paste error as it should mention CURLOPT_MIMEPOST

jvelilla replied:
yes, fixed.

jocelyn commented on Src/library/cURL/interface/curl_easy_externals_i.e:
```diff
@@ -115,6 +115,16 @@ feature -- Command
 		end
 
+	setopt_mime (a_curl_handle: POINTER; a_opt: INTEGER; a_mime: POINTER)
+			-- Declared as curl_easy_setopt().
```
Not sure if this is a typo or not about "curl_easy_setopt()"
but it is, then the previous features have the same typo.

jvelilla replied:
Not sure, all the features calling the external C function used the same convention, maybe to indicate that was C function, I've updated all the comments.

jocelyn commented on Src/examples/cURL/smtp_mime/application.e:
```diff
@@ -0,0 +1,528 @@
+			l_message.replace_substring_all ("$TO", l_to)
+			l_message.replace_substring_all ("$CC", l_cc)
+			l_message.replace_substring_all ("$FROM", l_from)
+			if curl_easy.is_dynamic_library_exists then
```
each time I see this `is_dynamic_library_exists`  I find the name bad .. should be `dynamic_library_exists` or even `dynamic_library_available` 
but I guess too late to change.

jocelyn commented on Src/examples/cURL/smtp_mime/application.e:
```diff
@@ -0,0 +1,528 @@
+				mime := curl_externals.curl_mime_init (curl_handle)
+
+				    -- The inline part is an alterative proposing the html and the text versions of the e-mail.
+    			alt	:= curl_externals.curl_mime_init (curl_handle)
```
bad indentation

jvelilla replied:
fixed

jocelyn commented on Src/examples/cURL/smtp_mime/application.e:
```diff
@@ -0,0 +1,528 @@
+feature {NONE} -- Initialization
+
+	make
+			-- Run application.
```
maybe you can find a better comment.

jvelilla replied:
done

jvelilla committed:
Updated comments remove extra parenthesis when wrapping C external functions.
Fixed bad comments.
Updated ecf for mime type example.
Fixed identation in the application example, add a meaningful comment.

jocelyn commented on Src/library/cURL/interface/curl_externals_i.e:
```diff
@@ -253,9 +354,158 @@ feature {NONE} -- C externals
+			"[
+				#ifdef CURL_STATICLIB
+				(FUNCTION_CAST(void , (curl_mime *)) $a_api)
+										((curl_mime *)$a_curl_handle);
```
bad indentation

jocelyn commented on Src/library/cURL/interface/curl_externals_i.e:
```diff
@@ -253,9 +354,158 @@ feature {NONE} -- C externals
+			exists_mimepart: not a_curl_handle.is_default_pointer
+		external
+				"C inline use <curl/curl.h>"
+			alias
```
bad indentation for the whole alias block

jvelilla replied:
Fixed

jocelyn commented on Src/library/cURL/interface/curl_easy_externals_i.e:
```diff
@@ -355,7 +365,7 @@ feature {NONE} -- C externals
 	c_getinfo (a_api: POINTER; a_curl_handle: POINTER; a_opt: INTEGER; a_data: POINTER): INTEGER
 			-- C implementation of `curl_easy_getinfo'.
-			-- Declared as curl_easy_setopt ().
+			-- Declared as curl_easy_setopt .
```
is it really "Declared as curl_easy_setopt"?? setopt ?
in fact I don't understand this comment here, and elsewhere.

jocelyn commented on Src/examples/cURL/smtp_mime/application.e:
```diff
@@ -0,0 +1,524 @@
+		do
+			io.put_string ("Eiffel cURL SMTP example showing how to send e-mails with mime.")
+			io.put_new_line
+			print ("%NLibCurl Veriosn:" + (create {CURL_OPT_CONSTANTS}).libcurl_version.out)
```
typo "Veriosn"...

jocelyn commented on Src/examples/cURL/smtp_mime/application.e:
```diff
@@ -0,0 +1,524 @@
+﻿note
+	description: "[
+					cURL https://curl.haxx.se/libcurl/c/smtp-mail.html
```
remove a few left indentations

jocelyn reviewed (approved):
a few things to fix, but LGTM

jvelilla committed:
Updated branch

jvelilla committed:
Fixed bad identations and a few typos.
